csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/AssemblyPath.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Semmle.Util.Logging;
+
+namespace Semmle.Extraction.CSharp.DependencyFetching
+{
+    /// <summary>
+    /// Used to represent a path to an assembly or a directory containing assemblies
+    /// and a selector function to determine which files to include, when indexing the assemblies.
+    /// </summary>
+    internal sealed class AssemblyPath(string p, Func<string, bool> includeFileName)
```

Comment:
Maybe `AssemblyLookupLocation` is a somewhat better name for this, as it's not necessarily an assembly path that we're storing here.